src/components/addMovie/AddMovie.jsx

```diff
@@ -1,70 +1,130 @@
-import React, { useState } from 'react';
+import React from 'react';
 import Form from 'react-bootstrap/Form';
 import Button from 'react-bootstrap/Button';
 import { genresOptions, labels, text } from '../../data.js';
 import { useDispatch } from 'react-redux';
 import { createMovie } from 'service/index.js';
+import { Formik } from "formik";
+import * as yup from "yup";
+import { FormTextField, FormSelectField } from "components/form-field";
 
 const AddMovie = ({ handleClose }) => {
     const { titleLabel, releaseDateLabel, selectDateLabel, movieUrlLabel, genreLabel, selectGenreLabel, overviewLabel, runtimeLabel, hereLabel } = labels;
     const { resetTxt, submitTxt } = text;
     
-    const [title, setTitle] = useState("");
-    const [release_date, setReleaseDate] = useState("");
-    const [poster_path, setMovieUrl] = useState("");
-    const [genres, setGenre] = useState([]);
-    const [overview, setOverview] = useState("");
-    const [runtime, setRuntime] = useState(0);
-
     const dispatch = useDispatch();
 
     const createMovieFunc = (params) => {
         createMovie(dispatch, params, () => {
-            handleClose()
+            handleClose();
         })
     }
 
+    const schema = yup.object({
+        title: yup.string().required(),
+        release_date: yup.date().required(),
+        poster_path: yup.string().url().required(),
+        genres: yup.array().nullable(),
+        overview: yup.string().required(),
+        runtime: yup.number().max(120).required()
+    });
+
     return (
-        <>
-            <Form.Group controlId="movieTitle">
-                <Form.Label>{titleLabel}</Form.Label>
-                <Form.Control type="text" placeholder={titleLabel} onChange={e => setTitle(e.target.value)} />
-            </Form.Group>
-            <Form.Group controlId="movieReleaseDate">
-                <Form.Label>{releaseDateLabel}</Form.Label>
-                <Form.Control type="text" placeholder={selectDateLabel} onChange={e => setReleaseDate(e.target.value)} />
-            </Form.Group>
-            <Form.Group controlId="movieUrl">
-                <Form.Label>{movieUrlLabel}</Form.Label>
-                <Form.Control type="text" placeholder={`${movieUrlLabel} ${hereLabel}`} onChange={e => setMovieUrl(e.target.value)} />
-            </Form.Group>
-            <Form.Group controlId="movieGenre">
-                <Form.Label>{genreLabel}</Form.Label>
-                <Form.Control as="select" onChange={e => setGenre([e.target.value])}>
-                    <option>{selectGenreLabel}</option>
+        <Formik
+          validationSchema={schema}
+          onSubmit={createMovieFunc}
+          initialValues={{
```

Comment:
Please save such objects as separate variable, because you could easily have state object with 20> fields, assigning it directly into property will look awful.
Also please consider providing prop types if you have any in your project, because its importance (know which kind of data is expected in an object) in such ^ cases is one of the highest among other use-cases.